lms/djangoapps/certificates/views/support.py

```diff
@@ -35,22 +34,25 @@
 log = logging.getLogger(__name__)
 
 
-def require_certificate_permission(func):
+def require_certificate_permission(permission):
     """
     View decorator that requires permission to view and regenerate certificates.
     """
-    @wraps(func)
-    def inner(request, *args, **kwargs):
-        if request.user.has_perm(VIEW_ALL_CERTIFICATES, 'global'):
-            return func(request, *args, **kwargs)
-        else:
+    def inner(func):
+        @wraps(func)
+        def wrapper(request, *args, **kwargs):
+            if request.user.has_perm(VIEW_ALL_CERTIFICATES, 'global') and permission == 'view':
+                return func(request, *args, **kwargs)
+            elif (request.user.has_perm(GENERATE_ALL_CERTIFICATES, 'global') and
+                  permission == 'regenerate'):
+                return func(request, *args, **kwargs)
```

Comment:
I'd probably just change this so that the `permission` is one of `VIEW_ALL_CERTIFICATES` or `GENERATE_ALL_CERTIFICATES` (or any future cert permission), so that the code looks like:

```
if request.user.has_perm(permission, 'global'):
    return func(request, *args, **kwargs)
```

That way, we don't have to modify this decorator every time we add a new type of cert permission.